kernel/fs/iso9660.inc

```diff
@@ -736,3 +736,1044 @@ cd_compare_name:
+        ; copy real name
+        movzx   ecx, byte[ebx - 1]
+        cmp     [ebp + ISO9660.type_encoding], 0
+        jnz     .ucs2
+        cmp     eax, 2
+        je      .ascii2utf16
+@@:
+        sub     ecx, 1 ; CF + ZF
+        jbe     @f
+        movsd
+        cmp     byte[esi], ';'
+        jne     @b
+@@:
+        mov     byte[edi], 0
+        jmp     .end_copy
+;-----------------------------------------------------------------------------
+.ascii2utf16:
+        lodsb
+        call    ansi2uni_char
+        stosw
+        cmp     byte[esi], ';'
+        je      @f
+        loop    .ascii2utf16
+@@:
+        mov     word[edi], 0
+        jmp     .end_copy
+;-----------------------------------------------------------------------------
+.ucs2:
+        shr     ecx, 1
+        cmp     eax, 1
+        ja      .ucs2utf16
+        ; convert ucs2 to ascii
+@@:
+        lodsw
+        cmp     ax, 0x3B00
+        je      @f
+        xchg    al, ah
+        call    uni2ansi_char
+        stosb
+        loop    @b
+@@:
+        mov     byte[edi], 0
+        jmp     .end_copy
+;-----------------------------------------------------------------------------
+.ucs2utf16:
+        cmp     edx, 2 ; utf16 encoding
+        jne     .ucs2utf8
+        ; convert ucs2 to utf16LE
+@@:
+        lodsw
+        cmp     ax, 0x3B00
+        je      @f
+        xchg    al, ah
+        stosw
+        loop    @b
+@@:
+        mov     word[edi], 0
+        jmp     .end_copy
+;-----------------------------------------------------------------------------
+.ucs2utf8:
+        ; convert ucs2 to utf8
+        mov     ebx, ecx
+        shl     ecx, 1
+@@:
+        lodsw
+        cmp     ax, 0x3B00
+        je      @f
+        xchg    ah, al
+        call    UTF16to8
+        dec     ebx
+        jnz     @b
+@@:
+        mov     byte[edi], 0
+;-----------------------------------------------------------------------------
+.end_copy:
+        pop     eax
+        pop     edi
+        pop     ebx
+        add     edi, 264
+        cmp     eax, 2
+        jb      .skip
+        add     edi, 520-264
+.skip:
+        movzx   ecx, byte[ebx]
+        add     ebx, ecx
+
+        test    ebx, 2048
+        jnz     .next_sector
+
+        mov     eax, ebx
+        and     eax, not -2048
+        cmp     eax, [esp + 4]
+        jb      .new_file
+
+        mov     eax, [esp + 12]
+        cmp     eax, [esp + 8]
+        jb      .new_file
+
+        jmp     .end_loop
+
+.next_sector:
+        inc     dword[esp]
+        sub     dword[esp + 4], 2048
+        ja      .read_sector
+.end_loop:
+        mov     ecx, [esp + 28]
+        mov     ebx, [esp + 12]
+        mov     [ecx + 4], ebx
+        mov     esi, [esp + 16]
+        mov     [ecx + 8], esi
+        mov     esi, [esp + 8] ; max count
+        ; free buffer
+        add     esp, 8*4
+        call    kernel_free
+
+        xor     eax, eax
+        cmp     ebx, esi
+        je      @f
+        mov     eax, ERROR_END_OF_FILE
```

Comment:
PR error code corrections are already in the master. Fix it `EEOF`.

Reply:
fixed